clearnode/app_session_service.go

```diff
@@ -339,8 +354,195 @@ func (s *AppSessionService) GetAppSessions(participantWallet string, status stri
 	return sessions, nil
 }
 
-// verifyQuorum loads an open AppSession, verifies signatures meet quorum
-func verifyQuorum(tx *gorm.DB, appSessionID string, rpcSigners map[string]struct{}) (AppSession, map[string]int64, error) {
+func getParticipantAllocations(db *gorm.DB, appSession AppSession, sessionAccountID AccountID) (map[string]map[string]decimal.Decimal, error) {
+	participantAllocations := make(map[string]map[string]decimal.Decimal)
+
+	for _, participant := range appSession.ParticipantWallets {
+		participantAllocations[participant] = make(map[string]decimal.Decimal)
+
+		ledger := GetWalletLedger(db, common.HexToAddress(participant))
+		balances, err := ledger.GetBalances(sessionAccountID)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, balance := range balances {
+			if !balance.Amount.IsZero() {
+				participantAllocations[participant][balance.Asset] = balance.Amount
+			}
+		}
+	}
+
+	return participantAllocations, nil
+}
+
+func (s *AppSessionService) handleOperateIntent(tx *gorm.DB, params *SubmitAppStateParams, appParticipants map[string]int64, sessionAccountID AccountID, appSessionBalance map[string]decimal.Decimal) error {
+	allocationSum := map[string]decimal.Decimal{}
+	for _, alloc := range params.Allocations {
+		if alloc.Amount.IsNegative() {
+			return RPCErrorf("negative allocation: %s for asset %s", alloc.Amount, alloc.AssetSymbol)
+		}
+
+		walletAddress := GetWalletBySigner(alloc.ParticipantWallet)
+		if walletAddress == "" {
+			walletAddress = alloc.ParticipantWallet
+		}
+
+		if err := validateAppParticipant(walletAddress, appParticipants); err != nil {
+			return err
+		}
+
+		userAddress := common.HexToAddress(walletAddress)
+		ledger := GetWalletLedger(tx, userAddress)
+		balance, err := ledger.Balance(sessionAccountID, alloc.AssetSymbol)
+		if err != nil {
+			return RPCErrorf("failed to get session balance for asset %s", alloc.AssetSymbol)
+		}
+
+		if err := ledger.Record(sessionAccountID, alloc.AssetSymbol, balance.Neg()); err != nil {
+			return RPCErrorf("failed to debit session: %w", err)
+		}
+		if err := ledger.Record(sessionAccountID, alloc.AssetSymbol, alloc.Amount); err != nil {
+			return RPCErrorf("failed to credit participant: %w", err)
+		}
+
+		if !alloc.Amount.IsZero() {
+			allocationSum[alloc.AssetSymbol] = allocationSum[alloc.AssetSymbol].Add(alloc.Amount)
+		}
+	}
+
+	if err := verifyAllocations(appSessionBalance, allocationSum); err != nil {
+		return RPCErrorf("incorrect operate request: non-zero allocation sum delta")
```

Comment:
```suggestion
		return RPCErrorf("incorrect operate request: non-zero allocations sum delta")
```

Changing for consistency.
Actually, for this to be better visible, we can extract errors to constants and declare them neat each other.

Reply:
67846776b253f0dc628f8f99374eb3ab0c9e9668